scripts/check-redact-log

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+# This script checks if there is plain (not redacted) log in TiKV.
+
+set -e
+
+if grep -r -n -P '(?<!hex_)encode_upper' -C 1 --color=always --include \*.rs --exclude hex.rs --exclude-dir target . ; then
+    echo "To print user data into info logs or error messages, use log_wrappers::Value() instead of hex::encode_upper. The former will respect \`security.redact-info-log\` config and filter out user data from info log if needed. Otherwise, use \`tikv_util::hex_encode_upper\` to get around the lint error. See https://github.com/tikv/tikv/pull/9250 for more information." && false
```

Comment:
It takes a lot of scrolling to find the `false` here, putting it on the next line would help. I think it is more straightforward to `exit 1`. I would also output the message to stderr.

On Mac `-e` is available instead of `-P` and it doesn't seem to support lookbehind. But that seems okay if this is always ran on CI.

Reply:
Yes, I spent lots of efforts to write the rule into this simple regex, and it seems to use some features only available in GNU grep...

This command was planned to be used as a one-liner in Makefile, and was later separated into this file. I'll change that to use `exit` and make it more readable soon.